spring-integration-core/src/test/java/org/springframework/integration/config/InvalidChannelWithCapacityAndQueueParserTests.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2002-2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.config;
+
+import org.hamcrest.Matchers;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import org.springframework.beans.factory.parsing.BeanDefinitionParsingException;
+import org.springframework.context.support.ClassPathXmlApplicationContext;
+
+/**
+ * @author Manuel Jordan
+ */
+public class InvalidChannelWithCapacityAndQueueParserTests {
+
+	@Rule
+	public ExpectedException exception = ExpectedException.none();
+
+	@Test
+	public void testQueueCapacityAndRefIllegal() throws Exception {
+		exception.expect(BeanDefinitionParsingException.class);
+		exception.expectMessage(Matchers.containsString("'capacity' attribute is not allowed"));
```

Comment:
Even if it is test class, I prefer to add `this.` just to work out the good habit, when I come to the `main` classes.